src/peer_connection.cpp

```diff
@@ -5460,23 +5460,22 @@ namespace libtorrent
 	int peer_connection::wanted_transfer(int channel)
 	{
 		TORRENT_ASSERT(is_single_thread());
-		shared_ptr<torrent> t = m_torrent.lock();
```

Comment:
are you sure this line is not necessary? (it has the secondary effect of keeping the torrent alive)

Reply:
I was thinking of it, but didn't find any practical reason why torrent object's lifetime must be prolonged during this call.